src/libsystemd-network/sd-dhcp6-client.c

```diff
@@ -231,6 +234,20 @@ int sd_dhcp6_client_set_iaid(sd_dhcp6_client *client, uint32_t iaid) {
         return 0;
 }
 
+int sd_dhcp6_client_set_fqdn(
+                sd_dhcp6_client *client,
+                const char *fqdn) {
+
+        assert_return(client, -EINVAL);
+
+        /* Refuse hostnames that neither qualify as DNS nor as Linux hostnames */
+        if (fqdn &&
+            !(hostname_is_valid(fqdn, false) && dns_name_is_valid(fqdn) > 0))
```

Comment:
you check here that the name must be valid as both linux and DNS hostname. That doesn't match what the comment says, and also isn't really what we want here, is it? (the comment appears right, but the code doesn't)

Reply:
According to discussion #7308. I agree that we should be using "or" here. If anything then dns_name_is_valid does not behave as expected.